### PR TITLE
feat: support testnet 4

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out Leather build 1b455a5 — [Extension build](https://github.com/leather-io/extension/actions/runs/11406449129), [Test report](https://leather-io.github.io/playwright-reports/feat-testnet4), [Storybook](https://feat-testnet4--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat-testnet4)

Closes LEA-1406

Adds testnet4 option

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]

File: src/app/features/add-network/network-form-fields.tsx [hunk 1]
```diff
@@ -66,15 +72,20 @@ export function NetworkFormFields({ isEditNetworkMode }: NetworkFormFieldsProps)
     [setFieldValue]
   );
 
-  function setNetworkUrls(value: BitcoinNetworkModes) {
+  function setNetworkUrls(value: BitcoinNetwork | BitcoinNetworkModes) {
```
Comment (alter-eggo) [inline src/app/features/add-network/network-form-fields.tsx]: just `BitcoinNetwork` no?

**Review by alter-eggo — COMMENTED** [review]
package.json deleted

File: src/app/common/utils.ts [hunk 2]
```diff
@@ -60,16 +55,23 @@ export function makeStacksAddressExplorerLink({
...
   txid,
   bitcoin: { bitcoinUrl, bitcoinNetwork },
 }: MakeBitcoinTxExplorerLinkArgs) {
   switch (bitcoinNetwork) {
     case 'mainnet':
-    case 'testnet':
-      return `https://mempool.space/${
-        bitcoinNetwork !== 'mainnet' ? bitcoinNetwork + '/' : ''
-      }tx/${txid}`;
+      return 'https://mempool.space/tx/' + txid;
```
Comment (pete-watters) [inline src/app/common/utils.ts]: `https://mempool.space/` should probably be a reused `const`

File: src/app/features/add-network/network-form-fields.tsx [hunk 3]
```diff
@@ -66,15 +72,20 @@ export function NetworkFormFields({ isEditNetworkMode }: NetworkFormFieldsProps)
...
     switch (value) {
       case 'mainnet':
         setStacksUrl('https://api.hiro.so');
         setBitcoinUrl(BITCOIN_API_BASE_URL_MAINNET);
         break;
       case 'testnet':
+      case 'testnet3':
+        setStacksUrl('https://api.testnet.hiro.so');
```
Comment (pete-watters) [inline src/app/features/add-network/network-form-fields.tsx:81]: `'https://api.testnet.hiro.so'` is repeated often

**Review by pete-watters — APPROVED** [review]
Nice work 👍  

I added some suggestions about keeping the URL code DRY but it's not a big deal as the code is already like that anyway

**Pull request merged**